piptools/utils.py

```diff
@@ -48,13 +48,33 @@ def make_install_requirement(name, version, extras, constraint=False):
     )
 
 
+def is_url_requirement(ireq, repository=None):
+    """
+    Finds if a requirement is a URL
+    """
+    if (
+        not ireq.link
+        or "files.pythonhosted.org" in ireq.link.url
+        or is_file_url(ireq.link)
+    ):
+        return False
+    if repository is not None and hasattr(repository, "finder"):
+        if any(
+            index_url in ireq.link.url for index_url in repository.finder.index_urls
+        ):
+            return False
+    return True
+
+
 def format_requirement(ireq, marker=None, hashes=None):
     """
     Generic formatter for pretty printing InstallRequirements to the terminal
     in a less verbose way than using its `__str__` method.
     """
     if ireq.editable:
         line = "-e {}".format(ireq.link)
+    elif is_url_requirement(ireq):
```

Comment:
Could you add a test like `test_format_requirement_url` in the `tests/test_utils.py` and cover that branch?